crates/wdk-build/src/resource_compile.rs

```diff
@@ -0,0 +1,217 @@
+use std::{
+    env,
+    fs,
+    path::{Path, PathBuf},
+    process::Command,
+};
+use cargo_metadata::MetadataCommand;
+
+// function to generate and compile RC file
+pub fn generate_and_compile_rcfile(include_paths: Vec<PathBuf>, rc_exe_rootpath: String) {
+    // Initialize an empty vector to store modified include arguments
+    let mut includeargs: Vec<String> = Vec::new();
+
+    // Iterate over each include path
+    for include_path in include_paths {
+        // Convert the include path to a string
+        if let Some(include_str) = include_path.to_str() {
+            // Append "/I" and the include path to the modified vector
+            includeargs.push("/I".to_string());
+            includeargs.push(include_str.to_string());
+        } else {
+            println!("Non-Unicode path is not supported: {:?}", include_path);
+        }
+    }
+
+    let (companyname, copyright, productname) = get_packagemetadatadetails();
+    let (productversion, description, fileversion, name) = get_packagedetails();
+    getandset_rcfile(
+        companyname, 
+        copyright, 
+        productname, 
+        productversion,
+        description, 
+        fileversion, 
+        name, 
+        &includeargs,
+        rc_exe_rootpath,
+    );
+}
+// function to get and set RC File with package metadata
+fn getandset_rcfile(
+    companyname: String, 
+    copyright: String, 
+    productname: String, 
+    productversion:String, 
+    description:String, 
+    fileversion:String, 
+    name:String, 
+    includeargs: &Vec<String>, 
+    rc_exe_rootpath:String,
+) {
+    println!("Set and create rc file... ");
+    let rcfile_path = "resources.rc";
+    if fs::metadata(&rcfile_path).is_ok() {
+        // File exists, so let's remove it
+        if let Err(err) = fs::remove_file(&rcfile_path) {
+            eprintln!("Error deleting file: {}", err);
+        } else {
+            println!("File deleted successfully!");
+        }
+    } else {
+        println!("File does not exist.");
+    }
+
+    let ver_filetype = "VFT_DRV";
+    let ver_filesubtype = "VFT2_DRV_SYSTEM";
+    let ver_originalfilename = "VER_INTERNALNAME_STR";
+
+    // Create the RC file content
+    let rc_content = format!(
+        r#"#include <windows.h>
+#include <ntverp.h>
```

Comment:
Is there an MSDN doc that this template is pulled from? Are these all required? optional?